owncloudApp/src/main/java/com/owncloud/android/ui/errorhandling/ErrorMessageAdapter.kt

```diff
@@ -272,35 +262,190 @@ class ErrorMessageAdapter {
+            res: Resources
+        ): String? {
+            val f = Formatter(res)
+
+            if (operation is UploadFileOperation)
+                return f.format(
+                    R.string.uploader_upload_failed_content_single,
+                    operation.fileName
+                )
+            if (operation is DownloadFileOperation)
+                return f.format(
+                    R.string.downloader_download_failed_content,
+                    File(operation.savePath).name
+                )
+            if (operation is RemoveFileOperation) return f.format(R.string.remove_fail_msg)
+            if (operation is RenameFileOperation) return f.format(R.string.rename_server_fail_msg)
+            if (operation is CreateFolderOperation) return f.format(R.string.create_dir_fail_msg)
+            if (operation is CreateShareViaLinkOperation || operation is CreateShareWithShareeOperation)
+                return f.format(R.string.share_link_file_error)
+            if (operation is RemoveShareOperation) return f.format(R.string.unshare_link_file_error)
+            if (operation is UpdateShareViaLinkOperation || operation is UpdateSharePermissionsOperation)
+                return f.format(R.string.update_link_file_error)
+            if (operation is MoveFileOperation) return f.format(R.string.move_file_error)
+            if (operation is SynchronizeFolderOperation)
+                return f.format(
+                    R.string.sync_folder_failed_content,
+                    File(operation.folderPath).name
+                )
+            if (operation is CopyFileOperation) return f.format(R.string.copy_file_error)
+            // if everything else failes
+            return if (result.isSuccess)
+                f.format(android.R.string.ok)
+            else
+                f.format(R.string.common_error_unknown)
+        }
+
+        /**
+         * TODO Remove methods above when new architecture is completed
+         */
+
+        /**
+         * Return an internationalized user message corresponding to an operation result
+         * and the operation performed.
+         *
+         * @param resultCode            ResultCode of a [RemoteOperation] performed.
+         * @param resultException       Exception of a [RemoteOperation] performed.
+         * @param operationType         [OperationType] performed.
+         * @param resources             Reference to app resources, for i18n.
+         * @return User message corresponding to 'result' and 'operation'
+         */
+        fun getResultMessage(
+            resultCode: RemoteOperationResult.ResultCode?,
+            resultException: Exception?,
+            operationType: OperationType,
+            resources: Resources
+        ): String {
+            val f = Formatter(resources)
+
+            when (resultCode) {
+                RemoteOperationResult.ResultCode.FORBIDDEN -> {
+                    if (operationType == OperationType.UPLOAD_FILE)
```

Comment:
why don't you use when instead of so many if else statements???

Reply:
Done, guess I will eventually get used to use `when` instead of `if else`